src/main/kotlin/graphics/scenery/primitives/Cylinder.kt

```diff
@@ -18,63 +18,95 @@ import kotlin.math.sqrt
  * @param[segments] Number of segments in latitude and longitude.
  */
 
-class Cylinder(var radius: Float, var height: Float, var segments: Int) : Mesh("cylinder") {
+class Cylinder(var radius: Float, var height: Float, var segments: Int, fillCaps: Boolean = false, smoothSides: Boolean = false) : Mesh("cylinder") {
```

Comment:
Please add the `@JvmOverloads` annotation to the constructor (it'll look a bit different afterwards, search for the annotation in the scenery code base) - otherwise Java callees will have to always use the full signature, as Java doesn't do default args.